src/js/Router.js

```diff
@@ -37,13 +33,35 @@ export default class Router extends String {
      * @return {String}
      */
     toString() {
+        return this.resolve({name: this._name, params: this._params});
+    }
+
+    /**
+     * Con
+     * @param {string} name
+     * @param {Object|string} [params]
+     * @param {Object} [query]
+     * @returns {string}
+     */
+    resolve({name, params, query}) {
+        if (!name) {
```

Comment:
As explained this is the toString logic abstracted to not depend on the instance route and params, so that any route can be passed to it